src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,159 +1,390 @@
+
+        Yields
+        ------
+        astropy.table.Table
+            Chunks of the input table
+        """
+        # If using entire table as one chunk, just yield the whole table
+        if self.chunk_size is None:
+            yield table
+            return
+
+        if self.chunking_mode == "events":
+            yield from self._get_event_chunks(table, chunk_shift, effective_chunk_size)
+        elif self.chunking_mode == "time":
+            yield from self._get_time_chunks(table, chunk_shift, effective_chunk_size)
+
+    def _check_table_length(self, table, effective_chunk_size):
+        """Check if the table length is sufficient for at least one chunk."""
+        if self.chunking_mode == "events":
+            if len(table) < effective_chunk_size:
+                raise ValueError(
+                    f"The length of the provided table ({len(table)}) "
+                    f"is insufficient to meet the required statistics "
+                    f"for a single chunk of size ({effective_chunk_size})."
+                )
+        elif self.chunking_mode == "time":
+            times = table["time"]
+            total_duration = (times[-1] - times[0]).to_value("s")
+            if total_duration < effective_chunk_size:
+                raise ValueError(
+                    f"The total duration of the provided table"
+                    f"({total_duration} seconds) is insufficient "
+                    f"to meet the required statistics for a single chunk "
+                    f"of size ({effective_chunk_size} seconds)."
+                )
+
+    def _get_event_chunks(self, table, chunk_shift, effective_chunk_size):
+        """Generate chunks based on event count."""
+        # Calculate the range step: Use chunk_shift if provided, otherwise use chunk_size
+        step = chunk_shift or effective_chunk_size
+
+        # Generate chunks that do not overflow
+        for i in range(0, len(table) - effective_chunk_size + 1, step):
+            yield table[i : i + effective_chunk_size]
+
+        # If chunk_shift is None, ensure the last chunk is of size chunk_size, if needed
+        if chunk_shift is None and len(table) % effective_chunk_size != 0:
+            yield table[-effective_chunk_size:]
+
+    def _get_time_chunks(self, table, chunk_shift, effective_chunk_size):
+        """Generate chunks based on time intervals."""
+        times = table["time"]
+        start_time = times[0]
+        end_time = times[-1]
+        total_duration = (end_time - start_time).to_value("s")
+
+        # Calculate time step: Use chunk_shift if provided, otherwise use chunk_size
+        time_step = chunk_shift if chunk_shift is not None else effective_chunk_size
+
+        current_time = start_time
+        while True:
+            chunk_end = current_time + effective_chunk_size * u.s
+            # Check if chunk end exceeds our data range (with 0.1s tolerance for floating point)
+            if chunk_end > end_time + 0.1 * u.s:
```

Comment:
I'd assume that this tolerance depends on your use case. I'd pass it as an argument in the `__call__()`